core/src/common.cpp

```diff
@@ -207,6 +218,178 @@ namespace irods::http
+
+	auto res{irods::base64_encode(reinterpret_cast<const unsigned char*>(_view.data()),
+								  _view.size(),
+								  reinterpret_cast<unsigned char*>(encoded_data.data()),
+								  &max_size_plus_null_term)};
+	if (res) {
+	  log::debug("{}: Yikes!!!", __func__);
+	}
+	return encoded_data;
+  }
+
+	auto create_host_field(boost::urls::url_view _url, std::string_view _port) -> std::string {
+		if ((_port.compare("443") == 0 && _url.scheme_id() == boost::urls::scheme::https) ||
+			(_port.compare("80") == 0 && _url.scheme_id() == boost::urls::scheme::http)) {
+			return _url.host();
+		}
+		return fmt::format("{}:{}", _url.host(), _port);
+	}
+
+	auto create_oidc_request(boost::urls::url_view _url) -> beast::http::request<beast::http::string_body> {
+		constexpr auto version_number{11};
+		beast::http::request<beast::http::string_body> req{beast::http::verb::post, _url.path(), version_number};
+
+		const auto port{get_port_from_url(_url)};
+
+		req.set(beast::http::field::host, create_host_field(_url, *port));
+		req.set(beast::http::field::user_agent, irods::http::version::server_name);
+		req.set(beast::http::field::content_type,
+		        "application/x-www-form-urlencoded");
+		req.set(beast::http::field::accept, "application/json");
+
+		if (const auto& g_config{irods::http::globals::oidc_configuration()}; g_config.contains("client_secret")) {
+			const auto format_bearer_token{[](std::string_view _client_id, std::string_view _client_secret){
+				auto encode_me{fmt::format("{}:{}", encode(_client_id), encode(_client_secret))};
+				return safe_base64_encode(encode_me);
+			}};
+
+			const auto& client_id{g_config.at("client_id").get_ref<const std::string&>()};
+			const auto& client_secret{g_config.at("client_secret").get_ref<const std::string&>()};
+			const auto auth_string{fmt::format("Basic {}", format_bearer_token(client_id, client_secret))};
+
+			req.set(beast::http::field::authorization, auth_string);
+		}
+
+		return req;
+	}
+
+    auto hit_intro_endpoint(std::string _encoded_body) -> nlohmann::json
+	{
+		const auto introspection_endpoint{
+			irods::http::globals::oidc_endpoint_configuration().at("introspection_endpoint").get_ref<const std::string&>()};
+
+		const auto parsed_uri{boost::urls::parse_uri(introspection_endpoint)};
+
+		if (parsed_uri.has_error()) {
+			log::error(
+				"{}: Error trying to parse introspection_endpoint [{}]. Please check configuration.", __func__, introspection_endpoint);
+			return {{"error", "bad endpoint"}};
+		}
+
+		const auto url{*parsed_uri};
+		const auto port{get_port_from_url(url)};
+
+		// Addr
+		net::io_context io_ctx;
+		auto tcp_stream{irods::http::transport_factory(url.scheme_id(), io_ctx)};
+		tcp_stream->connect(url.host(), *port);
+
+		const auto format_bearer_token{[](std::string_view _client_id, std::string_view _client_secret){
+		  auto encode_me{fmt::format("{}:{}", encode(_client_id), encode(_client_secret))};
+		  return safe_base64_encode(encode_me);
+		}};
```

Comment:
I've seen this twice now. Consider moving into a function.

Reply:
This lambda and some variables below it aren't used, since this is already taken care of in another function, `create_oidc_request(...)`.

Reply:
Cleaned up.